### PR TITLE
LUCENE-9990: gradle7 support

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

File: gradle/java/javac.gradle [hunk 1]
```diff
@@ -16,6 +16,15 @@
  */
 
 // Configure Java project defaults.
+java {
+  toolchain {
+    languageVersion = JavaLanguageVersion.of(11)
+  }
+}
+
+tasks.withType( JavaCompile ).configureEach {
```
Comment (dweiss) [inline gradle/java/javac.gradle:25]: Add the issue reference you mentioned here. The entire workaround thing should be best moved under a separate file in gradle/hacks.

File: gradle/testing/alternative-jdk-support.gradle [hunk 2]
```diff
@@ -18,60 +18,60 @@
...
-
-    // Javadoc compilation.
-    def javadocExecutable = altJvm.jdk.get().javadocExecutable.asFile
-    tasks.matching { it.name == "renderJavadoc" || it.name == "renderSiteJavadoc" }.all {
-      dependsOn ":altJvmWarning"
-      executable = javadocExecutable
-    }
-  }
-}
+//JavaInstallationRegistry registry = extensions.getByType(JavaInstallationRegistry)
```
Comment (dweiss) [inline gradle/testing/alternative-jdk-support.gradle:21]: I understand it's a PoC but a plan is needed how to enable what worked previously - pointing at an arbitrary JVM home and have it be the default for compilation, tests, etc. In the current form the patch removes things that are currently used to separate the gradle vs. lucene JVM toolchains - we do need this separation (for example to rule out early access JVM problems stemming from gradle itself vs. lucene).

File: gradle/wrapper/gradle-wrapper.jar.version [hunk 3]
```diff
@@ -1 +1 @@
-6.8.3
+7.0.2
```
Comment (dweiss) [inline gradle/wrapper/gradle-wrapper.jar.version:1]: You'll also need to update the hash of the wrapper?

Comment (gautamworah96) [reply gradle/wrapper/gradle-wrapper.jar.version:1]: I checked https://services.gradle.org/distributions/ and the SHA256 of the gradle-7.0.2-wrapper.jar.sha256 has remained the same. It was the same for the 6.6.1 to 6.8.3 transition as well

File: gradle/java/javac.gradle (hunk 1 above)
Comment (gautamworah96) [reply gradle/java/javac.gradle:25]: Sure. gradle/hacks makes sense. I'll also follow up on the ticket to see if there is a better way to solve this.

File: gradle/testing/alternative-jdk-support.gradle (hunk 2 above)
Comment (gautamworah96) [reply gradle/testing/alternative-jdk-support.gradle:21]: Sure.
Gradle 7 has something similar to what we have already setup today i.e., using an env variable to send in the location of the JVM. I can get it running and then add instructions in lucene/help/jvms.txt

Comment (dweiss) [reply gradle/testing/alternative-jdk-support.gradle:21]: Just to be clear: I don't know how it works exactly but what we want is not one jvm to run gradle *and* lucene - we want the ability to separate these.

Comment (gautamworah96) [reply gradle/testing/alternative-jdk-support.gradle:21]: Understood :) The gradle website says this (under improvements in Gradle 7): 

`Additionally, you may want to build a project using a Java version that is not supported for running Gradle.` 

so I am optimistic that we should be able to run Gradle and Lucene on different Java versions.

Sorry for the delay in the response! I will try my best to find some time soon and get this running

Comment (dweiss) [reply gradle/testing/alternative-jdk-support.gradle:21]: No worries at all. I think Palantir's plugin is the showstopper here too.

**Comment by gautamworah96** [comment]
I gave this PR another shot (since the Palantir plugin has been patched in v2.0.0 for Gradle 7 support), but had some new issues come up. The good news, I *think* that using the `-Porg.gradle.java.installations.paths` command line param points Gradle to use that specific JDK for building and running the project. The bad news, since `JavaInstallationRegistry` is now deprecated, the build fails in multiple places (some that use the Java version to add specific JVM params, and others where we use the plugin to get the Java command to generate some javadoc). As of right now, I am just trial-and-erroring some code to see what works. 

Some WIP code is pushed [here](https://github.com/gautamworah96/lucene/pull/new/LUCENE-9990).

**Pull request closed without merging**